File: SupportPrefenceDemo/app/src/main/java/edu/cs4730/supportprefencedemo/AdvancedPreferenceFragment.java
package edu.cs4730.supportprefencedemo;

import android.os.Bundle;

import androidx.preference.PreferenceFragmentCompat;

/**
 * A holder for the advanced preferences xml.
 */
public class AdvancedPreferenceFragment extends PreferenceFragmentCompat {

    @Override
    public void onCreatePreferences(Bundle bundle, String rootKey) {
        setPreferencesFromResource(R.xml.preferencesadvanced, rootKey);
    }
}

File: SupportPreference2Demo_kt/app/src/main/java/edu/cs4730/supportpreference2demo_kt/AdvancedPreferenceFragment.kt
package edu.cs4730.supportpreference2demo_kt

import android.os.Bundle
import androidx.preference.PreferenceFragmentCompat

/**
 * A holder for the advanced preferences xml.
 */
class AdvancedPreferenceFragment : PreferenceFragmentCompat() {
    override fun onCreatePreferences(bundle: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.preferencesadvanced, rootKey)
    }
}
